web3/pm.py

```diff
@@ -204,7 +207,7 @@ def _num_release_ids(self, package_name: str) -> int:
         pass
 
     @abstractmethod
-    def deploy_new_instance(cls, w3):
+    def deploy_new_instance(cls, w3: Web3) -> 'ERC1319Registry':
```

Comment:
This is *ok* but there are two models that I think improve this:

In the case where you want to stick to a well defined API:

```python
class ThingAPI(ABC):
    ...

class Thing(ThingAPI):
    @classmethod
    def make_thing(cls) -> ThingAPI:
        ...
```

This pattern helps enforce the defined API and is only the right solution in some cases.

In all of the other cases:

```python
T = TypeVar("T")

class Thing:
    @classmethod
    def make_thing(cls: Type[T]) -> T:
        ...
```

This ensures that in the case of subclassing, the return type isn't locked into the base class.